Libraries/Components/Button.js

```diff
@@ -273,11 +278,14 @@ class Button extends React.Component<ButtonProps> {
         buttonStyles.push({backgroundColor: color});
       }
     }
-    const accessibilityState = {};
+    const accessibilityState =
+      disabled === true
+        ? {...this.props.accessibilityState, disabled}
+        : this.props.accessibilityState;
+
     if (disabled) {
       buttonStyles.push(styles.buttonDisabled);
       textStyles.push(styles.textDisabled);
-      accessibilityState.disabled = true;
```

Comment:
@kacieb brought up a great point -- this PR doesn't seem to be fixing accessibility for talkback but rather adding an ability for accessibilityState -- I tested on an Android device and did notice that talkback does announce "disabled" for the button in your example where you solely use `accessibilityState`. 

That raises the question of what was the original issue reporting? Was it reporting that `accessibilityState={{disabled..` wasn't working -- which is indeed what your PR fixes -- but the fact that the property didn't even exist before has me puzzled

Reply:
cc @blavalla who reported the issue for clarification -- relevant to our discussion on disabled vs. accessibilityState.disabled

Reply:
I'm going to hold off landing this PR until we have some clarity about what the original issue is.

Reply:
Noted! 
Link to the accessibility issue: https://github.com/facebook/react-native/issues/30840

Reply:
@lunaleaps, the original issue was that setting `accessibilityState: disabled` alone was not actually disabling the button for accessibility.

This snack (https://snack.expo.io/d1Ikg7nqo) from #30840 shows the issue well.

Reply:
To clarify further, the issue was mostly that it is weird that you can actually set conflicting states for accessibility and non-accessibility users.  Basically, there is nothing stopping someone from setting `disabled="true"` `accessibilityState={{disabled: false}}` or vice-versa. 

If we are going to explicitly allow this though, they should at least work as expected, with the accessibilityState: disabled actually disabling the element for accessibility users.

Reply:
Ah right! `accessibilityState` is a View level prop so it should already exist on `<Button />`. My mistake! Okay that clarifies things! It sounds like this change is still worth pursuing -- to just sync the two values

Reply:
> If we are going to explicitly allow this though, they should at least work as expected, with the accessibilityState: disabled actually disabling the element for accessibility users.

@lunaleaps I explicitly tested `disabled="false"` and  `accessibilityState={{disabled: true}}` and it **announces** that button is **disabled** but it **does not disables** the **onPress**. I believe this needs to be done on the **View** level.

Having two different props is confusing. To disable the onPress we have to `explicitly` pass the `disabled={true}` prop.

Reply:
> and it announces that button is disabled but it does not disables the onPress. I believe this needs to be done on the View level.

Yea I think that's covered by https://github.com/facebook/react-native/issues/30943

Yea we can at least move forward with this to add both and we are discussing about consolidating the two props